### PR TITLE
add property tests

### PR DESCRIPTION
**Pull request by Sild**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java [hunk 1]
```diff
@@ -1674,14 +1395,202 @@ public void testEntitiesWildcardQuestionChar() throws Exception {
...
+        property.addTag("t1", "v1");
+        insertPropertyCheck(property);
+
+        PropertyQuery query = prepareSimplePropertyQuery(property.getType(), property.getEntity());
+        query.addKey("k1", null);
+        query.setExactMatch(true);
+
+        Response response = queryProperty(query);
+
+        assertEquals("Fail to execute property query", OK.getStatusCode(), response.getStatus());
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java]: Maybe extract these assertions  to separate method?

Comment (Sild) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java]: What exactly you suggest to extract? One assert row?

Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java]:
```java
assertEquals("Fail to execute property query", OK.getStatusCode(), response.getStatus());
final String given = response.readEntity(String.class);
final String expected = jacksonMapper.writeValueAsString(Collections.singletonList(property));
assertTrue("Property does not match to expected", compareJsonString(expected, given));
```

**Comment by Sild** [comment]
it's ok that 6 tests fail. Relative issue is #3352

File: src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java [hunk 2]
```diff
@@ -69,15 +61,13 @@ public void testFutureDateExactDefault() throws Exception {
...
-        queryObj.put("entity", property.getEntity());
-        queryObj.put("key", property.getKey());
+        PropertyQuery deleteQuery = new PropertyQuery();
+        deleteQuery.setType(property.getType());
+        deleteQuery.setEntity(property.getEntity());
 
-        assertEquals("Fail to execute delete query", OK.getStatusCode(), deleteProperty(queryObj).getStatus());
-        assertFalse("Property should be deleted", propertyExist(property));
+        assertEquals("Fail to execute delete query", OK.getStatusCode(), deleteProperty(deleteQuery).getStatus());
+        assertTrue("Property should be deleted", propertyExist(property));
```
Comment (GregoryKutuzov) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java]: Do you have a mistake?

File: src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java [hunk 3]
```diff
@@ -379,61 +473,311 @@ public void testExtraKeyPropNoKeyExactFalse() throws Exception {
...
+     * #2416
+     */
+    @Test
+    public void testKeyValueInteger() throws Exception {
+        final Property property = new Property("delete-type-28", "delete-entity28");
+        property.addTag("t1", "v1");
+        property.addKey("k1", "111");
+        insertPropertyCheck(property);
+
+        Map<String, Object> deleteQuery = new HashMap<>();
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java]: Why don't use PropertyQuery?

File: src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java [hunk 4]
```diff
@@ -379,61 +473,311 @@ public void testExtraKeyPropNoKeyExactFalse() throws Exception {
...
+     * #2416
+     */
+    @Test
+    public void testKeyValueBoolean() throws Exception {
+        final Property property = new Property("delete-type-29", "delete-entity29");
+        property.addTag("t1", "v1");
+        property.addKey("k1", "true");
+        insertPropertyCheck(property);
+
+        Map<String, Object> deleteQuery = new HashMap<>();
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java]: Why don't use PropertyQuery?

File: src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java [hunk 5]
```diff
@@ -157,71 +129,59 @@ public void testExactTrueWildcartKeyMatch() throws Exception {
...
-        }});
-        queryObj.put("exactMatch", true);
+        PropertyQuery query = prepareSimplePropertyQuery(property.getType(), "*");
+        query.setKey(property.getKey());
+        query.addKey("extra_key", "extra_key_val");
+        query.setExactMatch(true);
 
-        JSONAssert.assertEquals("[]", queryProperty(queryObj).readEntity(String.class), false);
+        String expected = jacksonMapper.writeValueAsString(Collections.emptyList());
+        Response response = queryProperty(query);
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java]: What about String actual instead?

File: src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java [hunk 6]
```diff
@@ -157,71 +129,59 @@ public void testExactTrueWildcartKeyMatch() throws Exception {
...
-            put("misskey", "misskey_value");
-        }});
-        queryObj.put("exactMatch", true);
+        PropertyQuery query = prepareSimplePropertyQuery(property.getType(), "*");
+        query.setKey(property.getKey());
+        query.addKey("extra_key", "extra_key_val");
+        query.setExactMatch(true);
 
-        JSONAssert.assertEquals("[]", queryProperty(queryObj).readEntity(String.class), false);
+        String expected = jacksonMapper.writeValueAsString(Collections.emptyList());
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyQueryTest.java]: Could we use "[]" instead?

File: src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java (hunk 4 above)
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/property/PropertyDeleteTest.java]: To check that database converts int to string.

**Pull request merged**